internal/common/normalattackstate.go

```diff
@@ -134,3 +140,146 @@ func Get5PercentN0Delay(activeChar *character.CharWrapper) int {
+		Core:        Core,
+		abilKey:     abilKey,
+		abilProcICD: abilProcICD,
+		abilICDKey:  abilICDKey,
+		abilTickSrc: 0,
+		abilHookSrc: 0,
+		delayFunc:   delayFunc,
+		summonFunc:  summonFunc,
+	}
+
+}
+
+func (w *NAHook) NAStateHook() {
+	w.Core.Events.Subscribe(event.OnAttack, func(args ...interface{}) bool {
+		//check if buff is up
+		if !w.c.StatusIsActive(w.abilKey) {
+			return false
+		}
+		w.abilHookSrc = w.Core.F
+		delay := w.delayFunc(w.Core.Player.ActiveChar())
+		w.Core.Log.NewEvent(fmt.Sprintf("%v delay on state change", w.abilName), glog.LogCharacterEvent, w.c.Index).
+			Write("delay", delay)
+		// This accounts for the delay in n0 timing needed to trigger
+
+		if delay > 0 {
+
+			w.Core.Tasks.Add(w.naStateDelayFuncGen(w.Core.F), delay)
+		} else {
+			// a delay of 0 will actually happen in the next frame, so a seperate conditional is used.
+
+			// Additionally, at the time that OnAttack/OnStateChange events are emitted, the state has not yet changed, so we cannot do an animation check.
+			if w.c.StatusIsActive(w.abilICDKey) {
+				w.Core.Log.NewEvent(fmt.Sprintf("%v did not trigger on state change", w.abilName), glog.LogCharacterEvent, w.c.Index).
+					Write("state", w.Core.Player.CurrentState()).
+					Write("icd", w.c.StatusExpiry(w.abilICDKey))
+			} else {
+				w.summonFunc()
+				w.c.AddStatus(w.abilICDKey, 60, true)
+				w.Core.Log.NewEvent(fmt.Sprintf("%v on state change", w.abilName), glog.LogCharacterEvent, w.c.Index).
+					Write("state", action.NormalAttackState).
+					Write("icd", w.c.StatusExpiry(w.abilICDKey))
+				w.abilTickSrc = w.Core.F
+				//use the hitlag affected queue for this
+				w.c.QueueCharTask(w.naTickerFunc(w.Core.F), w.abilProcICD) //check every `ICD`` frames
+			}
+		}
+
+		return false
+	}, fmt.Sprintf("%v animation check", w.abilName))
+}
+
+func (w *NAHook) naStateDelayFuncGen(src int) func() {
+	return func() {
+		//ignore if on ICD
+		if w.c.StatusIsActive(w.abilICDKey) || w.Core.Player.CurrentState() != action.NormalAttackState || w.abilHookSrc != src {
+			w.Core.Log.NewEvent(fmt.Sprintf("%v did not trigger on state change", w.abilName), glog.LogCharacterEvent, w.c.Index).
+				Write("state", w.Core.Player.CurrentState()).
+				Write("icd", w.c.StatusExpiry(w.abilICDKey)).
+				Write("ICD check", w.c.StatusIsActive(w.abilICDKey)).
+				Write("State check", w.Core.Player.CurrentState() != action.NormalAttackState).
+				Write("Source check", w.abilHookSrc != src)
+
+			return
+		}
+		//this should start a new ticker if not on ICD and state is correct
+		w.summonFunc()
+		w.c.AddStatus(w.abilICDKey, 60, true)
+		w.Core.Log.NewEvent(fmt.Sprintf("%v on state change", w.abilName), glog.LogCharacterEvent, w.c.Index).
+			Write("state", action.NormalAttackState).
+			Write("icd", w.c.StatusExpiry(w.abilICDKey))
+		w.abilTickSrc = w.Core.F
+		//use the hitlag affected queue for this
+		w.c.QueueCharTask(w.naTickerFunc(w.Core.F), w.abilProcICD) //check every `ICD` frames
+	}
+}
```

Comment:
This logic seems very similar/nearly identical to the `delay = 0` case in the subscription? can we reuse it in some way (have the different delayed check -> function that does status check + all the other necessary calls?)